core/lib/refinery/plugin.rb

```diff
@@ -1,5 +1,16 @@
 module Refinery
   class Plugin
+    META = {
```

Comment:
I think we should not introduce this feature in this PR, we already have `:hide_from_menu` (https://github.com/refinery/refinerycms/blob/c28e7afae816328280fb6f511094d8cbf100ebd0/core/lib/refinery/plugin.rb#L5) and 
`plugin_priority` (https://github.com/refinery/refinerycms/blob/master/core/lib/refinery/core/configuration.rb#L34) configs. 

I think we could just add the new config `plugin.icon` :

```ruby
# pages/lib/refinery/pages/engine.rb
Refinery::Plugin.register do |plugin|
  plugin.icon = 'icon-pages'
  plugin.pathname = root
  plugin.name = 'refinery_pages'
  plugin.menu_match = %r{refinery/page(_part|s_dialog)?s(/preview)?$}
  plugin.url = proc { Refinery::Core::Engine.routes.url_helpers.admin_pages_path }
end
```